protected/humhub/modules/content/widgets/WallCreateContentForm.php

```diff
@@ -50,6 +50,11 @@ class WallCreateContentForm extends Widget
      */
     protected $form = "";
 
+    /**
+     * @var bool Display menu above this form in order to select a content type like Post, Poll, Task and etc.
+     */
+    public $displayMenu = false;
+
     /**
```

Comment:
Would be great to have an additional property here, to completely disable the new feature.

e.g. 
```
public $enableContentTabs = true;
```

We could also rename `displayMenu` to  `displayContentTabs`.

Then it's possible via Configuration, to do something like this:

```
    'container' => [
        'definitions' => [
            'humhub\modules\content\widgets\WallCreateContentForm' => [
                'enableContentTabs' => false
            ],
        ],
    ],
```

Maybe we could also provide a callback possibility here to:
- Disable some content tabs
- Individually Sort Content Tabs/Menu Items

Reply:
@luke- Ok, I have renamed `displayMenu` to `displayContentTabs` in the commit https://github.com/humhub/humhub/pull/5827/commits/28fce66d5415f88e837c2a1d7842405e921e1605 and https://github.com/humhub/polls/pull/101/commits/f267a486620736db0e14d17610e311473465e86e.

But really the properly is used only to hide/show all menu tabs completely above the form, it is used on case when the form is loaded by ajax for Post and Poll, i.e. on ajax call we should not display the menu tabs again, only form should be displayed there.

Reply:
@yurabakhtin Okay, now I got it.

Wouldn't it be better to split it then?

- `humhub\modules\post\widgets\Form` extends `humhub\modules\content\widgets\WallCreateContentForm`.
- `humhub\modules\content\widgets\WallCreateContentForm` becomes abstract
- New `humhub\modules\content\widgets\WallCreateForm` as container class. Includes Tabs etc.

Reply:
@luke- I am not sure we need to split it, please read how it works now after commit https://github.com/humhub/humhub/pull/5827/commits/8d7e70a299761f9ef641a145b7f7d50105e8c693:

- In old version the post create form was rendered on home steam page by code `humhub\modules\post\widgets\Form::widget([...])`
- but in new version we should render top sorted form instead of the Post Form, so now the code was replaced with `humhub\modules\content\widgets\WallCreateContentForm:widget([...])` - https://github.com/humhub/humhub/pull/5827/commits/8d7e70a299761f9ef641a145b7f7d50105e8c693#diff-164af2fab55bf7dc8259870dcd57fb78b900ad9e1a121f5989cd026efddde697R31, there we find what form should be rendered as first form, so it may be Post or Poll form.
- The Post and Poll forms are extended from `humhub\modules\content\widgets\WallCreateContentForm`

---

If you need the split only in order to remove the property `public $displayContentTabs = false;` from the class `WallCreateContentForm` completely, then please confirm and I will try to implement this solution.
So new widget `humhub\modules\content\widgets\WallCreateForm` will be rendered as:

```
WallCreateContentMenu::widget([...]);
WallCreateContentForm::widget([...]);
```

right?

Reply:
@yurabakhtin  Thanks for the explanation, it is not clear for me why `The Post and Poll forms are extended from humhub\modules\content\widgets\WallCreateContentForm`.

`content\widgets\WallCreateContentForm` is more like a container, it holds the menu and renders inside `post\widgets\Form`. In your current solutions it's both. I think its much clearer if we split them: 

For example, I would find such a solution better separated from the code: 

## `content\widgets\WallCreateContentFormContainer`

formerly `content\widgets\WallCreateContentForm`. Is only the container, contains/integrates the tab menu. Contains the logic to preload the first available `content\widgets\AbstractWallCreateContentForm`. Checks if it should be displayed at all. (e.g. the user does not have permission to display a content).

This widget is placed above a Stream (Dashboard, Space, User Profile) to show the create input + tabs.
  
## e.g. `post\widgets\WallCreateContentForm`

Based on new `content\widgets\AbstractWallCreateContentForm`. Has mainy the code from the old `content\widgets\WallCreateContentForm` class for form submitting / content creation...

This class is not used directly in views. It's loaded by `WallCreateContentFormContainer` or returned by `WallStreamEntryWidget::createRoute`.

--- 

What do you think about the approach?

Reply:
@luke- Yes, I agree with your approach, I will implement this.

Reply:
@luke- Done in commit https://github.com/humhub/humhub/pull/5827/commits/ff9a8ae6a764bcd7b621d5ab1b1445b6f2e93e39.